### PR TITLE
vim: Fix matching brackets with 1 offset in normal mode

### LINKED ISSUE #17342: vim: wrong matching parenthesis highlighted
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

When in normal vim mode, the highlighted matching parenthesis is not the one under the cursor

### Environment

Zed: v0.151.0 (Zed Preview)
OS: Linux Wayland arch unknown
Memory: 31 GiB
Architecture: x86_64
GPU: Intel(R) Graphics (RPL-P) || Intel open-source Mesa driver || Mesa 24.2.1-arch1.1

### If applicable, add mockups / screenshots to help explain present your vision of the feature

image

### If applicable, attach your Zed.log file to this issue.

_No response_

### PR DESCRIPTION
**Pull request by jjyr**
Closes #17342

Release Notes:

- N/A

### CODE REVIEW

**Comment by jjyr** [comment]
@cla-bot check

File: crates/editor/src/highlight_matching_bracket.rs [hunk 1]
```diff
@@ -13,7 +14,11 @@ pub fn refresh_matching_bracket_highlights(editor: &mut Editor, cx: &mut ViewCon
         return;
     }
 
-    let head = newest_selection.head();
+    let mut head = newest_selection.head();
+    // Apply 1 offset for vim normal mode
+    if editor.cursor_shape == CursorShape::Block || editor.cursor_shape == CursorShape::Hollow {
+        head += 1;
+    }
```
Comment (maan2003) [inline crates/editor/src/highlight_matching_bracket.rs:21]: maybe this should be function on selection that take cursor shape as argument

same thing happen in symbol highlighting

**Comment by ConradIrwin** [comment]
I think just adding one causes the problem to happen in reverse on the open bracket (Though admittedly runs of open parentheses are less common). I've updated it in a way that I think fixes both. Does that work for you?

@maan2003 Unfortunately symbol highlighting is powered by the language server, and we can't make the same fix there as the protocol requires a single position instead of a range. We have to choose a position that is either to the left or the right of the current cursor; neither of which is quite perfect.

**Comment by ConradIrwin** [comment]
Also, thank you for picking this up! It's been grating on me for a long time but I didn't stop to think about it long enough.

**Pull request merged**